### PR TITLE
Support disable keep http connection alive

### PR DESCRIPTION
**Pull request by penghuazhou**
Support disable keep http connection alive

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*
- [x] README is updated with new configuration values *(if applicable)*

Relates to https://github.com/kedacore/keda/pull/3878
Relates to https://github.com/kedacore/keda/issues/3874

### CODE REVIEW

**Comment by tomkerkhove** [comment]
@penghuazhou Can you please fill in the PR template?

File: keda/templates/12-keda-deployment.yaml [hunk 1]
```diff
@@ -100,6 +100,10 @@ spec:
             - name: KEDA_HTTP_DEFAULT_TIMEOUT
               value: {{ .Values.http.timeout | quote }}
             {{- end }}
+            {{- if .Values.http.disableKeepAlive }}
```
Comment (tomkerkhove) [inline keda/templates/12-keda-deployment.yaml:103]: Let's use this approach which follows our typical way of configuring flags
```suggestion
            {{- if .Values.http.keepAlive.enabled }}
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]

File: keda/templates/12-keda-deployment.yaml (hunk 1 above)
Comment (JorTurFer) [reply keda/templates/12-keda-deployment.yaml:103]: Remember during the change that this requires changes in docs and code too ;)

Comment (penghuazhou) [reply keda/templates/12-keda-deployment.yaml]: @tomkerkhove @JorTurFer , as http transports is "DisableKeepAlives", so i transform the value in helm, please review again, thanks.
image

**Comment by penghuazhou** [comment]
> @penghuazhou Can you please fill in the PR template?

done

File: keda/templates/12-keda-deployment.yaml [hunk 2]
```diff
@@ -100,6 +100,10 @@ spec:
             - name: KEDA_HTTP_DEFAULT_TIMEOUT
               value: {{ .Values.http.timeout | quote }}
             {{- end }}
+            {{- if .Values.http.keepAlive.enabled }}
+            - name: KEDA_HTTP_DISABLE_KEEP_ALIVE
+              value: {{ .Values.http.keepAlive.enabled }}
+            {{- end }}
```
Comment (JorTurFer) [inline keda/templates/12-keda-deployment.yaml:106]: We can't do this as the current code has the opposite logic, when `.Values.http.keepAlive.enabled` is true, we disable it in the code.
I know that the code is already merged, so maybe we can handle this here adding some logic, for example:
```suggestion
            {{- if not( .Values.http.keepAlive.enabled) }}
            - name: KEDA_HTTP_DISABLE_KEEP_ALIVE
              value: "true"
            {{- end }}
```

And the same for the other manifest. WDYT?

**Review by JorTurFer — CHANGES_REQUESTED** [review]

File: keda/templates/12-keda-deployment.yaml (hunk 2 above)
Comment (penghuazhou) [reply keda/templates/12-keda-deployment.yaml]: @JorTurFer, done

File: keda/values.yaml [hunk 3]
```diff
@@ -254,6 +254,8 @@ priorityClassName: ""
 ## reasonable default
 http:
   timeout: 3000
+  keepAlive:
+    enabled: false
```
Comment (JorTurFer) [inline keda/values.yaml:258]: This should be `true` as default to maintaining current behaviour. When this is `true`, we don't add the env variable, so we don't disable the keep alive

```suggestion
    enabled: true
```

**Review by JorTurFer — APPROVED** [review]

**Comment by JorTurFer** [comment]
Can we merge this before the release @tomkerkhove ?

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**